web/src/app/admin/connectors/sharepoint/page.tsx

```diff
@@ -0,0 +1,250 @@
+import { usePublicCredentials } from "@/lib/hooks";
+import { AdminPageTitle } from "@/components/admin/Title";
+import { Card, Text, Title } from "@tremor/react";
+
+const MainSection = () => {
+  const { mutate } = useSWRConfig();
+  const {
+    data: connectorIndexingStatuses,
+    isLoading: isConnectorIndexingStatusesLoading,
+    error: isConnectorIndexingStatusesError,
+  } = useSWR<ConnectorIndexingStatus<any, any>[]>(
+    "/api/manage/admin/connector/indexing-status",
+    fetcher
+  );
+
+  const {
+    data: credentialsData,
+    isLoading: isCredentialsLoading,
+    error: isCredentialsError,
+    refreshCredentials,
+  } = usePublicCredentials();
+
+  if (
+    (!connectorIndexingStatuses && isConnectorIndexingStatusesLoading) ||
+    (!credentialsData && isCredentialsLoading)
+  ) {
+    return <LoadingAnimation text="Loading" />;
+  }
+
+  if (isConnectorIndexingStatusesError || !connectorIndexingStatuses) {
+    return <div>Failed to load connectors</div>;
+  }
+
+  if (isCredentialsError || !credentialsData) {
+    return <div>Failed to load credentials</div>;
+  }
+
+  const sharepointConnectorIndexingStatuses: ConnectorIndexingStatus<
+    SharepointConfig,
+    SharepointCredentialJson
+  >[] = connectorIndexingStatuses.filter(
+    (connectorIndexingStatus) =>
+      connectorIndexingStatus.connector.source === "sharepoint"
+  );
+
+  const sharepointCredential:
+    | Credential<SharepointCredentialJson>
+    | undefined = credentialsData.find(
+    (credential) => credential.credential_json?.aad_client_id
+  );
+
+  return (
+    <>
+      <Title className="mb-2 mt-6 ml-auto mr-auto">
+        Step 1: Provide Sharepoint credentials
+      </Title>
+      {sharepointCredential ? (
+        <>
+          <div className="flex mb-1 text-sm">
+            <Text className="my-auto">Existing Azure AD Client ID: </Text>
+            <Text className="ml-1 italic my-auto">
+              {sharepointCredential.credential_json.aad_client_id}
+            </Text>
+            <button
+              className="ml-1 hover:bg-hover rounded p-1"
+              onClick={async () => {
+                await adminDeleteCredential(sharepointCredential.id);
+                refreshCredentials();
+              }}
+            >
+              <TrashIcon />
+            </button>
+          </div>
+        </>
+      ) : (
+        <>
+          <Text className="mb-2">
+            To index Sharepoint, please provide
+            Azure AD client ID, Client Secret, and Directory ID.
+          </Text>
+          <Card className="mt-2">
+            <CredentialForm<SharepointCredentialJson>
+              formBody={
+                <>
+                  <TextFormField
+                    name="aad_client_id"
+                    label="Azure AD Client ID:"
+                  />
+                  <TextFormField
+                    name="aad_client_secret"
+                    label="Azure AD Client Secret:"
```

Comment:
Slight preference to have secret at the end. Probably best in general to have most generic to most sensitive. So Directory ID, Client ID, then Client Secret

Reply:
fixed